src/main/java/edu/ucla/library/iiif/auth/services/AccessCookieServiceImpl.java

```diff
@@ -134,24 +162,36 @@
 
     @Override
     public Future<JsonObject> decryptCookie(final String aCookieValue) {
-        final JsonObject decodedCookie = new JsonObject(new String(Base64.getDecoder().decode(aCookieValue)));
-        final byte[] encryptedCookieData = decodedCookie.getBinary(CookieJsonKeys.SECRET);
-        final byte[] nonce = decodedCookie.getBinary(CookieJsonKeys.NONCE);
-        final byte[] serializedCookieData;
         final JsonObject cookieData;
 
         try {
+            final JsonObject decodedCookie = new JsonObject(new String(Base64.getDecoder().decode(aCookieValue)));
+            final byte[] encryptedCookieData = decodedCookie.getBinary(CookieJsonKeys.SECRET);
+            final byte[] nonce = decodedCookie.getBinary(CookieJsonKeys.NONCE);
+            final byte[] serializedCookieData;
```

Comment:
I realize this is a bit more than you asked for @ksclarke but I think it's an improvement (?)

Reply:
Looks good to me.